src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -275,6 +275,20 @@ protected override ImmutableArray<PendingBranch> Scan(ref bool badRegion)
             Analyze(compilation, method, node, diagnostics, useMethodSignatureReturnType: false, useMethodSignatureParameterTypes: false, methodSignatureOpt: null, returnTypes: null, initialState: null, callbackOpt);
         }
 
+        internal static void Analyze(
+            CSharpCompilation compilation,
+            BoundAttribute attribute,
+            DiagnosticBag diagnostics)
+        {
+            if (attribute.Constructor is null ||
+                !compilation.SyntaxTrees.Contains(attribute.SyntaxTree))
```

Comment:
>!compilation.SyntaxTrees.Contains(attribute.SyntaxTree) [](start = 16, length = 55)

When is this true?

Reply:
During speculative binding via the sematic model [(GetSpeculativeSymbolInfo)]([https://docs.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.csharp.csharpextensions.getspeculativesymbolinfo?view=roslyn-dotnet#Microsoft_CodeAnalysis_CSharp_CSharpExtensions_GetSpeculativeSymbolInfo_Microsoft_CodeAnalysis_SemanticModel_System_Int32_Microsoft_CodeAnalysis_CSharp_Syntax_AttributeSyntax](https://docs.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.csharp.csharpextensions.getspeculativesymbolinfo?view=roslyn-dotnet#Microsoft_CodeAnalysis_CSharp_CSharpExtensions_GetSpeculativeSymbolInfo_Microsoft_CodeAnalysis_SemanticModel_System_Int32_Microsoft_CodeAnalysis_CSharp_Syntax_AttributeSyntax)_)

I considered putting the check in the attribute binder, rather than here?

---
In reply to: [264298961](https://github.com/dotnet/roslyn/pull/33816#discussion_r264298961) [](ancestors = 264298961)

Reply:
I think it deserves a comment.

Just curious: what fails if we remove this check?

---
In reply to: [264363456](https://github.com/dotnet/roslyn/pull/33816#discussion_r264363456) [](ancestors = 264363456,264298961)

Reply:
We fail in the constructor of nullable walker: we try and find the binder for the syntax tree, but there isn't one. 

---
In reply to: [264365353](https://github.com/dotnet/roslyn/pull/33816#discussion_r264365353) [](ancestors = 264365353,264363456,264298961)

Reply:
This is something I'm going to need to figure out in the public API.

---
In reply to: [264417374](https://github.com/dotnet/roslyn/pull/33816#discussion_r264417374) [](ancestors = 264417374,264365353,264363456,264298961)